### ISSUE TITLE
DBCollection.findAndRemove() throws an Exception instead of returning null

### ISSUE THREAD

**Issue by jgis** [foursquare/fongo#64]
If DBCollection.findAndRemove() does not find matching objects, it fails with a "java.lang.IllegalArgumentException: Don't know how to clone: null" in com.foursquare.fongo.impl.Util.clone().

The MongoDB driver instead returns null in this case.

Example:

``` java
DBObject query = new BasicDBObject("does", "notexist"); // matching objects do not exist in coll
coll.findAndRemove(query); // Fongo driver throws an exception, MongoDB driver returns null
```

Exception:
 java.lang.IllegalArgumentException: Don't know how to clone: null
    at com.foursquare.fongo.impl.Util.clone(Util.java:51)
    at com.mongodb.FongoDBCollection.findAndModify(FongoDBCollection.java:585)

findAndRemove() calls findAndModify(), which tries to return a clone of either the beforeObject or the afterObject:

```
if (returnNew){
  return Util.clone(afterObject);
} else {
  return Util.clone(beforeObject);
}
```

In the case of findAndRemove() with no matching objects, it tries to clone beforeObject, which is null.

Solution 1: handle nulls gracefully in com.foursquare.fongo.impl.Util.clone()
Solution 2: check for null in com.mongodb.FongoDBCollection.findAndModify()

**Comment by twillouer** [comment]
@hoffrocket if you plan to merge #62, take cares because findAndModify are modified very much

**Comment by hoffrocket** [comment]
fixed in 1.2.1-SNAPSHOT
